### PR TITLE
Optimal Control extension

### PR DESCRIPTION
**Pull request by lenasal**
optimal control methods for ALN, WC, Hopf and FHN models
example notebooks (deterministic and noisy)
tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -282,6 +287,29 @@ This will gives us a summary of the last generation and plots a distribution of
   <img src="https://github.com/neurolib-dev/neurolib/raw/master/resources/evolution_animated.gif">
 </p>
 
+### Optimal control
+The optimal control modules enables to compute efficient stimulation for your neural model. If you know how your output should look like, this module computes the optimal input. Detailes example notebooks can be found in the [optimal control example folder](https://github.com/lenasal/neurolib/tree/readme_update/examples/example-5-optimal-control). In optimal control computations, you trade precision with respect to a target against control strength. You can determine how much each contribution affects the results, by setting weights accordingly.
```
Comment (caglorithm) [inline README.md:291]: This link seems to point to the wrong repository.

File: README.md [hunk 2]
```diff
@@ -313,9 +341,16 @@ Cakan, C., Jajcay, N. & Obermayer, K. neurolib: A Simulation Framework for Whole
 
 Caglar Cakan (cakan@ni.tu-berlin.de)  
 Department of Software Engineering and Theoretical Computer Science, Technische Universität Berlin, Germany  
-Bernstein Center for Computational Neuroscience Berlin, Germany  
+Bernstein Center for Computational Neuroscience Berlin, Germany 
+
+For questions or comments on the optimal control module, please contact
+Lena Salfenmoser (lena.salfenmoser@tu-berlin.de)
+Department of Software Engineering and Theoretical Computer Science, Technische Universität Berlin, Germany  
+
```
Comment (caglorithm) [inline README.md:349]:
```suggestion
```

This should not be here. I suggest adding the citation to the OC paper below the OC example in the Readme.

File: README.md [hunk 3]
```diff
@@ -313,9 +341,16 @@ Cakan, C., Jajcay, N. & Obermayer, K. neurolib: A Simulation Framework for Whole
...
 
 ### Acknowledgments
 This work was supported by the Deutsche Forschungsgemeinschaft (DFG, German Research Foundation) with the project number 327654276 (SFB 1315) and the Research Training Group GRK1589/2.
 
+The optimal control module was developed and implemented by Lena Salfenmoser and Martin Krück, and supported by the Deutsche Forschungsgemeinschaft (DFG, German Research Foundation) with the project number 163436311 (SFB 910).
+
```
Comment (caglorithm) [inline README.md:355]:
```suggestion
The optimal control module was developed by Lena Salfenmoser and Martin Krück supported by the DFG project 163436311 (SFB 910).
```

This is a better place for Acknowledgment.

**Review by caglorithm — CHANGES_REQUESTED** [review]
Please adjust README, otherwise LGTM. I would say let's merge and fix bugs as they appear.

File: README.md (hunk 2 above)
Comment (lenasal) [reply README.md]: Removed. Not sure if citing the OC paper below the OC example makes sense, please have a look...

**Comment by lenasal** [comment]
thanks for reviewing, please have a look at the changes

File: README.md (hunk 3 above)
Comment (lenasal) [reply README.md]: I was basically copy-pasting what you wrote, just adapting names and project numbers... but I'm also fine with your suggestion

File: README.md [hunk 4]
```diff
@@ -95,6 +95,10 @@ neurolib/	 				# Main module
 ├── optimize/ 					# Optimization submodule
 	├── evolution/ 				# Evolutionary optimization
 	└── exploration/ 			# Parameter exploration
+├── control/optimal_control/			# (Optimal) control submodule
```
Comment (caglorithm) [inline README.md:98]:
```suggestion
├── control/optimal_control/			# Optimal control submodule
```

**Review by caglorithm — APPROVED** [review]

**Pull request merged**